src/plugins/ucx/ucx_backend.cpp

```diff
@@ -336,52 +341,75 @@ class nixlUcxBackendH : public nixlBackendReqH {
 void nixlUcxEngine::progressFunc()
 {
     using namespace nixlTime;
-    pthrActive = 1;
 
     vramApplyCtx();
 
+    {
+        std::unique_lock<std::mutex> lock(pthrActiveLock);
+        pthrActive = true;
+    }
+    pthrActiveCV.notify_one();
+
+    std::vector<pollfd> poll_fds;
+    for (auto &uw: uws) {
+        int fd;
+        if (ucp_worker_get_efd(uw->getWorker(), &fd) == UCS_OK)
+            poll_fds.push_back({fd, POLLIN, 0});
+        else
+            NIXL_ERROR << "Couldn't obtain fd for a worker";
+    }
+
     while (!pthrStop) {
-        int i;
-        for(i = 0; i < noSyncIters; i++) {
-            uw->progress();
+        bool made_progress = false;
+        for (auto &uw: uws)
+            while (uw->progress())
+                made_progress = true;
+
+        if (made_progress) {
```

Comment:
As mentioned in connection establishment, we can only have notifications on one worker (I suggest worker[0]).
So we can save few cycles calling `notifProgress()` only if progress was on worker[0].

Reply:
done